Adds back removed props and behavior back to Query Controls

draganescu opened the pull request:
This PR is fixing backward compatibility for plugins that use this component. The new behavior exists only if the new props are passed in.

Fixes #23285

## Description
In a PR that improved the Latest Posts block by adding support for filtering by multiple category the `QueryControls` component had a couple of public props removed and that caused issues with plugins which were implementing the component.

## How has this been tested?
Tested locally by using the component in both ways in the `LatestPosts` block.

## Types of changes
Added back removed code from the component so everything should behave as expected. To have the component provide the new multiple category selector, pass in the new props.

gziolo reviewed (approved):
It looks like it resolves backward compatibility issues.

What about React Native related props that were passed in the past to `CategorySelect`?

gziolo commented on packages/components/src/query-controls/index.js:
```diff
@@ -66,7 +69,16 @@ export default function QueryControls( {
+				categoriesList={ categoriesList }
+				label={ __( 'Category' ) }
+				noOptionLabel={ __( 'All' ) }
+				selectedCategoryId={ selectedCategoryId }
```
@hypest and @pinarol, there were some mobile-related props in the original commit:
https://github.com/WordPress/gutenberg/commit/e17d104ee3fe6eb8d81373226ab73096f59f9ba3

Do you think `MOBILE_CONTROL_PROPS` is still needed? Can someone double-check that?

hypest replied:
👋 @chipsnyder , can you have a look here since you worked on the Latest Posts block in the past? Thanks!

chipsnyder replied:
In the current state, we don't really need `MOBILE_CONTROL_PROPS` added back. 

Because of the original change we did end up removing categories and hiding them behind a `__DEV__` flag. https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/latest-posts/edit.native.js#L181

<details>
<summary>With this change, we could reenable categories on that block but there would be two issues we might want to consider.</summary>

The current Native implementation doesn't handle:
-  Multiselect
- For a long list of categories, the control that was used by default didn't handle scroll

We have https://github.com/wordpress-mobile/gutenberg-mobile/issues/2069 to track that work.
</details>

So I'd say we're good here but does raise the question again of if we want to keep categories disabled in mobile or not. 

ps thanks for checking with use @gziolo!

dmsnell commented:
Is this supposed to include all the thousands of lines in the diff? Looks like we might have had a merge issue when trying to update or create the PR?

draganescu commented:
There has been some problem with this PR ...

draganescu committed:
adds removed props and behavior back to Query Controls

fixing backward compat for plugins that use this component. the new behavior exists only if the new props are passed in.

ellatrix commented:
What's the status of this PR? Can it be merged?
